packages/indexer/src/lib/document-processor.ts

```diff
@@ -1,67 +1,55 @@
 import { type BaseLogger } from 'pino';
 import { getBlobNameFromFile } from './blob-storage.js';
+import { type ContentPage, type ContentSection, type Section } from './document.js';
+import { extractText, extractTextFromPdf } from './formats/index.js';
 
-export interface Document {
-  filename: string;
-  type: string;
-  category: string;
-  sections: Section[];
-}
-
-export interface Section {
-  id: string;
-  content: string;
-  category: string;
-  sourcepage: string;
-  sourcefile: string;
-  embedding?: number[];
-}
-
-export interface ContentPage {
-  content: string;
-  offset: number;
-  page: number;
-}
-
-export interface ContentSection {
-  content: string;
-  page: number;
-}
-
+const SENTENCE_ENDINGS = new Set(['.', '!', '?']);
+const WORD_BREAKS = new Set([',', ';', ':', ' ', '(', ')', '[', ']', '{', '}', '\t', '\n']);
 const MAX_SECTION_LENGTH = 1000;
 const SENTENCE_SEARCH_LIMIT = 100;
 const SECTION_OVERLAP = 100;
 
 export class DocumentProcessor {
-  constructor(private logger: BaseLogger) {}
+  formatHandlers = new Map<string, (data: Buffer) => Promise<ContentPage[]>>();
+
+  constructor(private logger: BaseLogger) {
+    this.registerFormatHandler('text/plain', extractText);
+    this.registerFormatHandler('text/markdown', extractText);
+    this.registerFormatHandler('application/pdf', extractTextFromPdf);
```

Comment:
this is great. I recommend making registerFormatHander public and call it in the indexer.ts so that it can be used as a generic processer.

Reply:
Will do that in a next PR